source/common/network/address_impl.cc

```diff
@@ -168,7 +168,7 @@ int InstanceBase::socketFromSocketType(SocketType socketType) const {
   RELEASE_ASSERT(fcntl(fd, F_SETFL, O_NONBLOCK) != -1, "");
 #endif
 
-  return fd;
+  return {fd, errno};
```

Comment:
In the `__APPLE__` case, I think that fcntl could be resetting errno, so you need to latch it earlier.

On the other hand, there's already a RELEASE_ASSERT that an error did not occur (fd != -1). So should this function even return errno?

Reply:
@ggreenway good catch! What do you suggest? Should we still keep the function signature as `Api::SysCallResult` for consistency, and to provide a way for future error handling?

Reply:
I'd say leave this function as returning an int, and make sure the docs mention that this will never fail.

Reply:
Addressed in dc0c613.